packages/list/index.js

```diff
@@ -82,219 +85,188 @@ export default class List extends Component {
 
-  getListItemClasses_(id, className) {
-    const {listItemClassList} = this.state;
-    return listItemClassList[id] ?
-      classnames(Array.from(listItemClassList[id]), className) :
-      className;
-  }
-
   get adapter() {
     return {
-      getListItemCount: () => Object.keys(this.listItems_).length,
-      getFocusedElementIndex: () => this.getListItemIndexOfTarget_(document.activeElement),
+      getListItemCount: () => this.listItemCount,
+      getFocusedElementIndex: () => {
+        let count = -1;
+        let index = -1;
+        React.Children.forEach(this.props.children, (child) => {
+          if (child.type.name === 'ListItem' && child === document.activeElement) {
+            index = count;
+          } else if (child.type.name === 'ListItem') {
+            count++;
+          }
+        });
+        return index;
+      },
       setAttributeForElementIndex: (index, attr, value) => {
         const {listItemAttributes} = this.state;
         attr = attr === 'tabindex' ? 'tabIndex' : attr;
-        const id = this.getListItemIdFromIndex_(index);
-        listItemAttributes[id][attr] = value;
+        if (!listItemAttributes[index]) {
+          listItemAttributes[index] = {};
+        }
+        listItemAttributes[index][attr] = value;
         this.setState({listItemAttributes});
       },
       removeAttributeForElementIndex: (index, attr) => {
         const {listItemAttributes} = this.state;
         attr = attr === 'tabindex' ? 'tabIndex' : attr;
-        const id = this.getListItemIdFromIndex_(index);
-        delete listItemAttributes[id][attr];
+        if (!listItemAttributes[index]) {
+          return;
+        }
+        delete listItemAttributes[index][attr];
         this.setState({listItemAttributes});
       },
       addClassForElementIndex: (index, className) => {
-        const {listItemClassList} = this.state;
-        const id = this.getListItemIdFromIndex_(index);
-        listItemClassList[id].add(className);
-        this.setState({listItemClassList});
+        const {listItemClassNames} = this.state;
+        if (!listItemClassNames[index]) {
+          listItemClassNames[index] = [];
+        }
+        listItemClassNames[index].push(className);
+        this.setState({listItemClassNames});
       },
       removeClassForElementIndex: (index, className) => {
-        const {listItemClassList} = this.state;
-        const id = this.getListItemIdFromIndex_(index);
-        listItemClassList[id].delete(className);
-        this.setState({listItemClassList});
-      },
-      focusItemAtIndex: (index) => {
-        const id = this.getListItemIdFromIndex_(index);
-        const listItem = this.listItems_[id];
-        if (listItem) {
-          listItem.focus();
+        const {listItemClassNames} = this.state;
+        if (!listItemClassNames[index]) {
+          return;
         }
+        const i = listItemClassNames[index].indexOf(className);
+        listItemClassNames[index].splice(i, 1);
```

Comment:
I think you need to check if `i<= -1`. If you do `listItemClassNames[index].splice(-1, 1)`, its going to delete something, which I believe is unintended.